api/channel.go

```diff
@@ -20,15 +20,15 @@ type CreateChannelData struct {
 	Name  string `json:"name"` // 2-100
 	Topic string `json:"topic,omitempty"`
 
-	Type discord.ChannelType `json:"type,omitempty"`
+	Type *discord.ChannelType `json:"type,omitempty"`
```

Comment:
This doesn't need to be a pointer: 0, the zero-value, is also the [default value](https://discord.com/developers/docs/resources/channel#channel-object-channel-types). Perhaps omitempty could be omitted.

Reply:
If it's the default, we might as well omit it. Omitting it also saves us a few bytes in the JSON payload, so I'd say we keep it.

Reply:
But pointer removed